src/ensembl/utils/database/unittestdb.py

```diff
@@ -57,6 +63,7 @@ class UnitTestDB:
             file (without headers) per table following the convention `<table_name>.txt` (optional).
         name: Name to give to the new database. If not provided, the last directory name of `dump_dir`
             will be used instead. In either case, the new database name will be prefixed by the username.
+        metadata: Use this metadata to create the schema instead of using an SQL schema file.
```

Comment:
I believe this argument is now more important than `dump_dir` (or equally important), so I would put it before or after `dump_dir`.

`tmp_path` docstring is missing.